src/app/legacy/containers/CpsRecommendations/RecommendationsPromoImage/index.jsx

```diff
@@ -13,11 +13,19 @@ const RecommendationsImage = ({ indexImage, lazyLoad }) => {
     return <ImagePlaceholder ratio={landscapeRatio} />;
   }
 
-  const { height, width, path, altText, copyrightHolder } = indexImage;
+  const {
+    height,
+    width,
+    path,
+    altText,
+    copyrightHolder,
+    optimoLocator,
+    optimoOriginCode,
+  } = indexImage;
 
   const ratio = (height / width) * 100;
-  const originCode = getOriginCode(path);
-  const locator = getLocator(path);
+  const originCode = getOriginCode(path) || optimoOriginCode;
```

Comment:
```suggestion
  const originCode = optimoOriginCode || getOriginCode(path);
```
Should this be the other way around? What does `getOriginCode` return in the event of an optimo `path` being provided?

This logic could also be encapsulated in the utility I suggested above, again so `RecommendationsPromoImage` is unware of the multiple types of promo data we need to support.

Reply:
I believe the path would be null in this case and getOriginCode will return null. As you suggested this logic is not encapsulated in the `extractPromoData({promo})`